### ISSUE TITLE
Add macos build to releases

### ISSUE THREAD

**Issue by kdheepak** [Canop/broot#278]
**Is your feature request related to a problem? Please describe.**

I'd like to download a pre-built release for macos using the same script I use for downloading it for linux machines. 

**Describe the solution you'd like**

Use github actions to build `broot` for all platforms and publish to releases.

**Additional context**

I use the following steps to build and distribute a rust binary.

https://github.com/kdheepak/taskwarrior-tui/blob/fad4e5f642e3120ff2e0894b234ac15cf74c839d/.github/workflows/ci.yml

Releases are created and uploaded to github when a new tag is created automatically.

![Screen Shot 2020-09-10 at 2 10 55 PM](https://user-images.githubusercontent.com/1813121/92797618-777fb900-f36f-11ea-83fb-8a5f1eaa55cc.png)

Additionally, I can download the binary using [`zinit`](https://github.com/zdharma/zinit) with just two lines in my `~/.zshrc`.

```zsh
zinit ice wait:2 lucid extract"" from"gh-r" as"command" mv"taskwarrior-tui* -> tt"
zinit load kdheepak/taskwarrior-tui
```

I currently don't have write permissions in `/usr/local` so I cannot use `brew` to install `broot`.

**Comment by Canop** [comment]
@kdheepak Can I assume you'll volunteer to test the built broot I'll do before I make it part of  the new release process (through a specific download) ?

**Comment by kdheepak** [comment]
Yup, I can test on macos! I was going to ask if you would like me to submit a PR for doing this through github actions as well. Let me know if you'd like that.

**Comment by Canop** [comment]
I don't want to use github actions. I must run the whole process of cross compiling all versions to check it works (it frequently doesn't) and I don't want to have git operations in the way.

I'll just update my cross-compiling script and let you check the result.

**Comment by kdheepak** [comment]
Got it! That'll be great! I will be able test it once you share it.

**Comment by Canop** [comment]
Turns out that building for MacOS (so using the mac OS SDK) on linux (using docker) isn't as easy as it seems...

**Comment by BirkhoffLee** [comment]
not related to this issue but worth saying: I'm looking forward to having this package on Homebrew :)

**Comment by Canop** [comment]
Official releases now include a macOS binary (aarch64-apple-darwin, in the release zip), built natively since 1.59. There's no x86_64 Mac binary; Homebrew and MacPorts cover that case. Closing.
